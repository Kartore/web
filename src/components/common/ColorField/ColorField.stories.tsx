import type { Meta, StoryObj } from '@storybook/react';

import { ColorField } from './index.ts';

const meta = {
  component: ColorField,
} satisfies Meta<typeof ColorField>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {},
};
