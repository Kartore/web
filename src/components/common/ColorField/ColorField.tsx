import type { FC } from 'react';
import { useRef } from 'react';

import type { AriaColorFieldProps } from '@react-aria/color';
import { useColorField } from '@react-aria/color';
import { useColorFieldState } from '@react-stately/color';

import { cn } from '~/utils/tailwindUtil';

export type ColorFieldProps = {
  className?: string;
} & AriaColorFieldProps;

export const ColorField: FC<ColorFieldProps> = ({ className, label, ...props }) => {
  const state = useColorFieldState(props);
  const ref = useRef(null);
  const {
    labelProps,
    inputProps,
    descriptionProps,
    errorMessageProps,
    isInvalid,
    validationErrors,
  } = useColorField(props, state, ref);
  return (
    <div className={cn('flex flex-row items-center justify-between', className)}>
      <label
        {...labelProps}
        className={cn('text-sm font-semibold text-gray-600', labelProps.className)}
      >
        {label}
      </label>
      <div
        className={
          'rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0'
        }
      >
        <input {...inputProps} ref={ref} className={cn('', inputProps.className)} />
      </div>
      {props.description && (
        <div {...descriptionProps} className={cn('text-xs', descriptionProps.className)}>
          {props.description}
        </div>
      )}
      {isInvalid && (
        <div {...errorMessageProps} className={cn('text-red text-xs', errorMessageProps.className)}>
          {validationErrors.join(' ')}
        </div>
      )}
    </div>
  );
};

ColorField.displayName = 'ColorField';
