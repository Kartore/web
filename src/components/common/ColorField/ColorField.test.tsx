import { describe } from 'vitest';
import { render } from '@testing-library/react';
import { composeStories } from '@storybook/react';

import * as Stories from './ColorField.stories';

const { Default } = composeStories(Stories);

describe('Component: ColorField', () => {
  describe('Snapshot', () => {
    it('Default', () => {
      const { asFragment } = render(<Default />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
