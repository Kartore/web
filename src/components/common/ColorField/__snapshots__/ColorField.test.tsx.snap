// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: ColorField > Snapshot > Default 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center justify-between"
  >
    <label
      class="text-sm font-semibold text-gray-600"
    />
    <div
      class="flex w-1/2 flex-row items-center gap-2 rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0"
    >
      <div
        class="h-4 w-4"
      >
        <button
          class="cursor-pointer transition-colors hover:bg-gray-100 active:bg-gray-300 h-full w-full border border-gray-300 focus:outline-0"
          type="button"
        >
          <div
            aria-label="transparent"
            aria-roledescription="color swatch"
            class="h-full w-full"
            id="react-aria-:re:"
            role="img"
            style="background-color: rgba(255, 255, 255, 0);"
          />
        </button>
      </div>
      <input
        autocomplete="off"
        autocorrect="off"
        class="min-w-10 flex-2 border-none focus-visible:outline-0"
        id="react-aria-:r0:"
        role="textbox"
        spellcheck="false"
        type="text"
        value=""
      />
      <input
        aria-labelledby="react-aria-:r8:"
        aria-roledescription="Number field"
        autocomplete="off"
        autocorrect="off"
        class="min-w-8 flex-1 border-none focus-visible:outline-0"
        id="react-aria-:r6:"
        inputmode="numeric"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</DocumentFragment>
`;
